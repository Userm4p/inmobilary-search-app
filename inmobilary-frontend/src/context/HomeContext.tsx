'use client';
import { IHomeContext } from '@/types/Home.types';
import { createContext } from 'react';

const homeContextInitialValues: IHomeContext = {
  isLoading: false,
  error: null,
  properties: [],
  getProperties: async () => {},
  handleInputChange: (e: React.ChangeEvent<HTMLInputElement>) => {
    e.preventDefault();
  },
  searchFilters: {},
};

export const HomeContext = createContext<IHomeContext>(homeContextInitialValues);
