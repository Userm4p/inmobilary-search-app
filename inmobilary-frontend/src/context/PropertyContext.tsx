import { IPropertyContext } from '@/types/Property.types';
import { createContext } from 'react';

const initialState: IPropertyContext = {
  getProperty: async () => {},
  isLoading: false,
  error: null,
  property: null,
};

export const PropertyContext = createContext<IPropertyContext>(initialState);
