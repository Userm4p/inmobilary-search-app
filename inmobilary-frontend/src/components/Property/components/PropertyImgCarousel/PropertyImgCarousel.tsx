import LazyImage from '@/components/LazyImage/LazyImage';
import {
  Carousel,
  CarouselContent,
  CarouselItem,
  CarouselNext,
  CarouselPrevious,
} from '@/components/ui/carousel';
import { PropertyContext } from '@/context/PropertyContext';
import React, { useContext } from 'react';

export const PropertyImgCarousel = () => {
  const { property } = useContext(PropertyContext);

  return (
    <div className="lg:w-[700px] md:w-[700px] sm:w-[600px] xs:w-[450px] xxs:w-[350px] w-[300px] px-12">
      <Carousel className="w-full rounded-none">
        <CarouselContent>
          {property?.images.map((img, index) => (
            <CarouselItem key={index}>
              <LazyImage
                src={img}
                alt={`Property Image ${index + 1}`}
                className="object-cover rounded-lg w-[600px]"
                width={800}
                height={500}
              />
            </CarouselItem>
          ))}
        </CarouselContent>
        <CarouselPrevious />
        <CarouselNext />
      </Carousel>
    </div>
  );
};
