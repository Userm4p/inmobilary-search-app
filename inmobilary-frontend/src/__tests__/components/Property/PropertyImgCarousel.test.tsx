import { render } from '@testing-library/react';
import { PropertyImgCarousel } from '@/components/Property/components/PropertyImgCarousel/PropertyImgCarousel';
import { PropertyContext } from '@/context/PropertyContext';

// Mock Embla Carousel
jest.mock('embla-carousel-react', () => ({
  __esModule: true,
  default: jest.fn(() => [
    jest.fn(), // carouselRef
    {
      canScrollPrev: jest.fn(() => false),
      canScrollNext: jest.fn(() => false),
      scrollPrev: jest.fn(),
      scrollNext: jest.fn(),
      on: jest.fn(),
      off: jest.fn(),
    }, // api
  ]),
  useEmblaCarousel: jest.fn(() => [
    jest.fn(), // carouselRef
    {
      canScrollPrev: jest.fn(() => false),
      canScrollNext: jest.fn(() => false),
      scrollPrev: jest.fn(),
      scrollNext: jest.fn(),
      on: jest.fn(),
      off: jest.fn(),
    }, // api
  ]),
}));

const mockProperty = {
  id: '1',
  name: 'Test Property',
  address: '123 Test St',
  price: 100000,
  ownerId: 'owner1',
  owner: {
    id: 'owner1',
    name: 'John Doe',
    address: '456 Owner St',
    photo: 'photo.jpg',
    birthday: '1990-01-01',
  },
  images: ['image1.jpg', 'image2.jpg', 'image3.jpg'],
  traces: [],
};

const mockContextValue = {
  isLoading: false,
  error: null,
  property: mockProperty,
  getProperty: jest.fn(),
};

describe('PropertyImgCarousel', () => {
  it('should render carousel with property images', () => {
    const { container } = render(
      <PropertyContext.Provider
        value={{
          property: mockContextValue.property,
          isLoading: mockContextValue.isLoading,
          error: mockContextValue.error,
          getProperty: mockContextValue.getProperty,
        }}
      >
        <PropertyImgCarousel />
      </PropertyContext.Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
