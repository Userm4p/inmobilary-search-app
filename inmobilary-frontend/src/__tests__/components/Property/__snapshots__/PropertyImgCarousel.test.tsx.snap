// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`PropertyImgCarousel should render carousel with property images 1`] = `
<div>
  <div
    class="lg:w-[700px] md:w-[700px] sm:w-[600px] xs:w-[450px] xxs:w-[350px] w-[300px] px-12"
  >
    <div
      aria-roledescription="carousel"
      class="relative w-full rounded-none"
      data-slot="carousel"
      role="region"
    >
      <div
        class="overflow-hidden"
        data-slot="carousel-content"
      >
        <div
          class="flex -ml-4"
        >
          <div
            aria-roledescription="slide"
            class="min-w-0 shrink-0 grow-0 basis-full pl-4"
            data-slot="carousel-item"
            role="group"
          >
            <div
              aria-busy="true"
              class="relative overflow-hidden bg-gray-200 object-cover rounded-lg w-[600px]"
              style="width: 100%; max-width: 800px;"
            >
              <div
                style="width: 100%; padding-bottom: 62.5%;"
              />
              <img
                alt="Property Image 1"
                class="absolute top-0 left-0 w-full h-full object-cover transition-opacity duration-500 ease-in-out opacity-0"
                decoding="async"
                loading="lazy"
                src="image1.jpg"
              />
              <div
                class="absolute inset-0 flex items-center justify-center text-sm text-gray-500"
              >
                <div
                  class="w-5 h-5 border-2 border-blue-500 border-t-transparent rounded-full animate-spin"
                />
              </div>
            </div>
          </div>
          <div
            aria-roledescription="slide"
            class="min-w-0 shrink-0 grow-0 basis-full pl-4"
            data-slot="carousel-item"
            role="group"
          >
            <div
              aria-busy="true"
              class="relative overflow-hidden bg-gray-200 object-cover rounded-lg w-[600px]"
              style="width: 100%; max-width: 800px;"
            >
              <div
                style="width: 100%; padding-bottom: 62.5%;"
              />
              <img
                alt="Property Image 2"
                class="absolute top-0 left-0 w-full h-full object-cover transition-opacity duration-500 ease-in-out opacity-0"
                decoding="async"
                loading="lazy"
                src="image2.jpg"
              />
              <div
                class="absolute inset-0 flex items-center justify-center text-sm text-gray-500"
              >
                <div
                  class="w-5 h-5 border-2 border-blue-500 border-t-transparent rounded-full animate-spin"
                />
              </div>
            </div>
          </div>
          <div
            aria-roledescription="slide"
            class="min-w-0 shrink-0 grow-0 basis-full pl-4"
            data-slot="carousel-item"
            role="group"
          >
            <div
              aria-busy="true"
              class="relative overflow-hidden bg-gray-200 object-cover rounded-lg w-[600px]"
              style="width: 100%; max-width: 800px;"
            >
              <div
                style="width: 100%; padding-bottom: 62.5%;"
              />
              <img
                alt="Property Image 3"
                class="absolute top-0 left-0 w-full h-full object-cover transition-opacity duration-500 ease-in-out opacity-0"
                decoding="async"
                loading="lazy"
                src="image3.jpg"
              />
              <div
                class="absolute inset-0 flex items-center justify-center text-sm text-gray-500"
              >
                <div
                  class="w-5 h-5 border-2 border-blue-500 border-t-transparent rounded-full animate-spin"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <button
        class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive border bg-background shadow-xs hover:bg-accent hover:text-accent-foreground dark:bg-input/30 dark:border-input dark:hover:bg-input/50 absolute size-8 rounded-full top-1/2 -left-12 -translate-y-1/2"
        data-slot="carousel-previous"
        disabled=""
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-arrow-left"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m12 19-7-7 7-7"
          />
          <path
            d="M19 12H5"
          />
        </svg>
        <span
          class="sr-only"
        >
          Previous slide
        </span>
      </button>
      <button
        class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive border bg-background shadow-xs hover:bg-accent hover:text-accent-foreground dark:bg-input/30 dark:border-input dark:hover:bg-input/50 absolute size-8 rounded-full top-1/2 -right-12 -translate-y-1/2"
        data-slot="carousel-next"
        disabled=""
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-arrow-right"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5 12h14"
          />
          <path
            d="m12 5 7 7-7 7"
          />
        </svg>
        <span
          class="sr-only"
        >
          Next slide
        </span>
      </button>
    </div>
  </div>
</div>
`;
